pkg/reader/ccip.go

```diff
@@ -625,89 +625,88 @@ func (r *ccipChainReader) GetRMNRemoteConfig(
-		for _, chain := range keys {
-			cfg := sourceConfigs[chain]
-			resp = resp.Append(consts.ContractNameOnRamp, chain, cfg.OnRamp)
-			// The local router is located in each source chain config. Add it once.
-			if len(resp[consts.ContractNameRouter][r.destChain]) == 0 {
-				resp = resp.Append(consts.ContractNameRouter, r.destChain, cfg.Router)
+		for _, sourceChain := range keys {
+			cfg := sourceConfigs[sourceChain]
+			resp = resp.Append(consts.ContractNameOnRamp, sourceChain, cfg.OnRamp)
+			// The local router is located in each source sourceChain config. Add it once.
+			if len(resp[consts.ContractNameRouter][chain]) == 0 {
+				resp = resp.Append(consts.ContractNameRouter, chain, cfg.Router)
 			}
 		}
 	}
 
 	// NonceManager and RMNRemote are in the offramp static config.
-	var staticConfig offRampStaticChainConfig
-	err = r.getDestinationData(
-		ctx,
-		r.destChain,
-		consts.ContractNameOffRamp,
-		consts.MethodNameOffRampGetStaticConfig,
-		&staticConfig,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("unable to lookup nonce manager (offramp static config): %w", err)
+	{
+		var staticConfig offRampStaticChainConfig
+		err := r.getDestinationData(
+			ctx,
+			chain,
+			consts.ContractNameOffRamp,
+			consts.MethodNameOffRampGetStaticConfig,
+			&staticConfig,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to lookup nonce manager (offramp static config): %w", err)
+		}
+		resp = resp.Append(consts.ContractNameNonceManager, chain, staticConfig.NonceManager)
+		resp = resp.Append(consts.ContractNameRMNRemote, chain, staticConfig.RmnRemote)
+		r.lggr.Infow("appending RMN remote contract address", "address", staticConfig.RmnRemote)
 	}
-	resp = resp.Append(consts.ContractNameNonceManager, r.destChain, staticConfig.NonceManager)
-	resp = resp.Append(consts.ContractNameRMNRemote, r.destChain, staticConfig.RmnRemote)
-	r.lggr.Infow("appending RMN remote contract address", "address", staticConfig.RmnRemote)
 
 	// FeeQuoter from the offRamp dynamic config.
-	var dynamicConfig offRampDynamicChainConfig
-	err = r.getDestinationData(
-		ctx,
-		r.destChain,
-		consts.ContractNameOffRamp,
-		consts.MethodNameOffRampGetDynamicConfig,
-		&dynamicConfig,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("unable to lookup fee quoter (offramp dynamic config): %w", err)
+	{
+		var dynamicConfig offRampDynamicChainConfig
+		err := r.getDestinationData(
+			ctx,
+			chain,
+			consts.ContractNameOffRamp,
+			consts.MethodNameOffRampGetDynamicConfig,
+			&dynamicConfig,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to lookup fee quoter (offramp dynamic config): %w", err)
+		}
+		resp = resp.Append(consts.ContractNameFeeQuoter, chain, dynamicConfig.FeeQuoter)
 	}
-	resp = resp.Append(consts.ContractNameFeeQuoter, r.destChain, dynamicConfig.FeeQuoter)
 
 	return resp, nil
 }
 
 func (r *ccipChainReader) DiscoverContracts(
 	ctx context.Context,
-	allChains []cciptypes.ChainSelector,
+	_ []cciptypes.ChainSelector, //Deprecated
```

Comment:
Why not remove it in this PR? No external calls for this AFAIK.

Reply:
done